signature/internal/rekor_set.go

```diff
@@ -0,0 +1,98 @@
+package internal
+
+import (
+	"encoding/json"
+)
+
+// This is the github.com/sigstore/rekor/pkg/generated/models.Hashedrekord.APIVersion for github.com/sigstore/rekor/pkg/generated/models.HashedrekordV001Schema.
+// We could alternatively use github.com/sigstore/rekor/pkg/types/hashedrekord.APIVERSION, but that subpackage adds too many dependencies.
+const HashedRekordV001APIVersion = "0.0.1"
```

Comment:
Can you open an issue with them to not pull in imports when defining types?

Reply:
The `APIVERSION` constant lives in a code-focused package that implements various aspects of that type.

The actual type definition lives in https://github.com/sigstore/rekor/blob/main/openapi.yaml , the package we import is Swagger-generated, so that’s already the smaller package. (In fact it’s the smaller one, which is why it doesn’t carry the constant.)

It might be possible to change the YAML file so that a constant is included in the generated model, I couldn’t immediately find a way (if the `version` field were an enum, it would reject unknown versions completely).

Really, if anything, we’d like to avoid using even the generated types package, because that already adds too many dependencies. So asking upstream to change packages we don’t want to use feels a bit unreasonable.